### PR TITLE
Implement adding card with bloc

### PR DESCRIPTION
**Pull request by NimeshPiyumantha**
[ITP-1746] [ITP-1749] [ITP-1752]

[ITP-1746]: https://fcodelabs.atlassian.net/browse/ITP-1746?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ITP-1749]: https://fcodelabs.atlassian.net/browse/ITP-1749?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ITP-1752]: https://fcodelabs.atlassian.net/browse/ITP-1752?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/ui/diary_home_page/diary_home_event.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@immutable
+abstract class DiaryHomeEvent {}
+
+class InputVisibleEvent extends DiaryHomeEvent {
+  InputVisibleEvent();
+}
+
+class SubmitDataEvent extends DiaryHomeEvent {
+  final String textFieldValue;
+  final String textFieldController;
```
Comment (KavinduDhananjaya) [inline lib/ui/diary_home_page/diary_home_event.dart]: change variable names.

File: lib/ui/diary_home_page/diary_home_view.dart [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+      username: entry.username,
+      description: entry.description,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    DiaryHomeBloc diaryHomeBloc = context.read<DiaryHomeBloc>();
+    
+    return BlocProvider(
```
Comment (KavinduDhananjaya) [inline lib/ui/diary_home_page/diary_home_view.dart]: dont use BlocProviders here.

File: lib/ui/diary_home_page/diary_home_event.dart (hunk 1 above)
Comment (NimeshPiyumantha) [reply lib/ui/diary_home_page/diary_home_event.dart]: done

File: lib/ui/sign_in_page/sign_in_page_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:training_app/ui/sign_in_page/sign_in_page_event.dart';
+import 'package:training_app/ui/sign_in_page/sign_in_page_state.dart';
+
+class SignInPageBloc extends Bloc<SignInPageEvent, SignInPageState> {
+  final TextEditingController textFieldController = TextEditingController();
```
Comment (KavinduDhananjaya) [inline lib/ui/sign_in_page/sign_in_page_bloc.dart]: remove controllers

**Pull request merged**